src/desktop/__tests__/e2e/Onboarding.test.js

```diff
@@ -0,0 +1,107 @@
+describe('Trinity desktop end-to-end', () => {
+    test(
+        'Trinity onboarding existing seed',
+        async () => {
+            const page = await global.__getBrowserPage('onboarding', false);
+
+            // Welcome
+            await page.waitForSelector('#to-seed-intro');
+            await page.click('#to-seed-intro');
+
+            // Seed intro
+            await page.waitForSelector('#to-seed-verify');
+            await page.click('#to-seed-verify');
+
+            // Seed input
+            await page.waitForSelector('.input__editable > div');
+            await page.focus('.input__editable > div');
+            await page.keyboard.type(
+                '999999999999999999999999999999999999999999999999999999999999999999999999999999999',
+            );
+            await page.click('#seed-verify-next');
+
+            // Account name
+            await page.waitForSelector('input');
+            await page.focus('input');
+            await page.keyboard.type('FOO');
+
+            await page.click('#account-name-next');
+
+            // Account password
+            await page.waitForSelector('.input__input:first-of-type input');
+            await page.focus('.input__input:first-of-type input');
+            await page.keyboard.type('FooBarBazFooBar');
+
+            await page.focus('.input__input:last-of-type input');
+            await page.keyboard.type('FooBarBazFooBar');
+
+            await page.click('#account-password-next');
+
+            // Done
+            await page.waitForSelector('#done-next');
+            const selector = await page.$('#done-next');
+
+            expect(selector).toBeTruthy();
+        },
+        990000,
+    );
+
+    test(
+        'Trinity onboarding new seed',
+        async () => {
+            const page = await global.__getBrowserPage('onboarding', false);
+
+            // Welcome
+            await page.waitForSelector('#to-seed-intro');
+            await page.click('#to-seed-intro');
+
+            // Seed intro
+            await page.waitForSelector('#to-seed-generate');
+            await page.click('#to-seed-generate');
+
+            // Seed generate
+            await new Promise((resolve) => setTimeout(resolve, 200));
+            for (let i = 1; i <= 10; i++) {
+                const selector = `.index__seed button:nth-child(${i})`;
+                await page.waitForSelector(selector);
+                await page.click(selector);
```

Comment:
This one seemed to fail twice out of the eight times I ran it. Is there anything we can do to help reduce the flakiness or is this an issue with the component itself?


```
 FAIL  __tests__/e2e/Onboarding.test.js (9.132s)
  Trinity desktop end-to-end
    ✓ Trinity onboarding existing seed (3496ms)
    ✕ Trinity onboarding new seed (3061ms)

  ● Trinity desktop end-to-end › Trinity onboarding new seed

    Node is either not visible or not an HTMLElement

      65 |                 const selector = `.index__seed button:nth-child(${i})`;
      66 |                 await page.waitForSelector(selector);
    > 67 |                 await page.click(selector);
         |                            ^
      68 |             }
      69 |             await page.click('#to-account-name');
      70 | 

      at ElementHandle._clickablePoint (node_modules/puppeteer/lib/JSHandle.js:199:13)
        -- ASYNC --
      at ElementHandle.<anonymous> (node_modules/puppeteer/lib/helper.js:110:27)
      at DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:367:18)
        -- ASYNC --
      at Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:110:27)
      at Page.click (node_modules/puppeteer/lib/Page.js:988:29)
      at Object.click (__tests__/e2e/Onboarding.test.js:67:28)
```

Reply:
It's Puppeteer's new page initialisation that causes the random delay. Will look into on how to reduce it.